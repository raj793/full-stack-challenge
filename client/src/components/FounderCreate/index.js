import React, { useState } from 'react';
import clsx from 'clsx';
import Api from '../../apis';
import './index.css';

const FounderCreate = ({onClose, companyId}) => {
  const apis = new Api();
  const formModel = {
    name: '',
    title: ''
  };

  const [formData, setFormData] = useState(formModel);
  const [errors, setErrors] = useState(formModel);
  const [isSaving, setSaving] = useState(false);

  const setFormValue = (e) => {
    const fd = {...formData};
    fd[e.target.name] = e.target.value;
    setFormData(fd);
  }

  const createNewCompany = async () => {
    if(formData.name && formData.title) {
      setSaving(true);
      await apis.createFounder({
        name: formData.name,
        title: formData.title,
        company: companyId
      });
      setSaving(false);
      onClose && onClose();
    }
    else {
      setErrors({
        name: !formData.name,
        title: !formData.title
      })
    }
  }

  return (
    <div className="form-container">
      <span className="heading">Add a founder</span>
      <input 
        className={clsx({["input-group-item"]: true, ["input-error"]: errors.name})} 
        name="name" 
        onChange={setFormValue} 
        value={formData.name} 
        placeholder=" Name *Required"
      />
      <input 
        className={clsx({["input-group-item"]: true, ["input-error"]: errors.title})}
        name="title"
        onChange={setFormValue}
        value={formData.titla}
        placeholder=" Title *Required"
      />
      <div className='save-button' onClick={createNewCompany}>{isSaving? "ADDING": "ADD"}</div>
    </div>
  );
}

export default FounderCreate;