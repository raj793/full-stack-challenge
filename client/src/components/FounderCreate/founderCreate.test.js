import React from 'react';
import renderer from 'react-test-renderer';
import FounderCreate from './index';

test('Test Founder Create snapshot', () => {
  const formData = {
    name: "Name",
    city: "City",
    state: "State",
    description: "Description"
  }
  const component = renderer.create(
    <FounderCreate />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});