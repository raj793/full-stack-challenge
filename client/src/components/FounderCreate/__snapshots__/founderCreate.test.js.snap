// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Founder Create snapshot 1`] = `
<div
  className="form-container"
>
  <span
    className="heading"
  >
    Add a founder
  </span>
  <input
    className="input-group-item"
    name="name"
    onChange={[Function]}
    placeholder=" Name *Required"
    value=""
  />
  <input
    className="input-group-item"
    name="title"
    onChange={[Function]}
    placeholder=" Title *Required"
  />
  <div
    className="save-button"
    onClick={[Function]}
  >
    ADD
  </div>
</div>
`;
